adam/batch_propagation_results.py

```diff
@@ -0,0 +1,271 @@
+        """Polls the job until the job completes.
+
+        Args:
+            max_wait_sec (int): the maximum time in seconds to run the wait.
+                Defaults to 60.
+            print_waiting (boolean): Whether to print the waiting status messages.
+                Defaults to False.
+
+        Returns:
+            str: the job status.
+        """
+
+        sleep_time_sec = 10.0
+        t0 = time.perf_counter()
+        status = self.check_status()
+        last_status = ''
+        count = 0
+        while status != 'COMPLETED':
+            if print_waiting:
+                if last_status != status:
+                    print(status)
+                    count = 0
+                if count == 40:
+                    count = 0
+                    print()
+                print('.', end='')
+            elapsed = time.perf_counter() - t0
+            if elapsed > max_wait_sec:
+                raise RuntimeError(
+                    f'Computation has exceeded desired wait period of {max_wait_sec} sec.')
+            last_status = status
+            time.sleep(sleep_time_sec)
+            status = self.check_status()
+
+    def get_results(self, force_update=True):
+        if force_update or self._results is None:
+            results = self._rp.get_results(self._job_uuid)
+            self._results = results
+
+        return self._results
+
+
+class MonteCarloResults(ApsResults):
+    """API for retrieving propagation results and summaries"""
+
+    class PositionOrbitType(Enum):
+        """The type of orbit position in relation to a target body."""
+        MISS = 'MISS'
+        CLOSE_APPROACH = 'CLOSE_APPROACH'
+        IMPACT = 'IMPACT'
+
+    @classmethod
+    def fromRESTwithRawIds(cls, rest, project_uuid, job_uuid):
+        results_processor = ApsRestServiceResultsProcessor(rest, project_uuid)
+        return MonteCarloResults(results_processor, job_uuid)
+
+    def __init__(self, results_processor, job_uuid):
+        ApsResults.__init__(self, results_processor, job_uuid)
+        self._detailedOutputs = None
+        self._summary = None
+
+    def get_summary(self, force_update=False):
+        """Get the propagation results summary.
+
+        Args:
+            force_update(boolean): True if calling this method should be re-executed,
+                otherwise False (default).
+
+        Returns:
+            summary (MonteCarloSummary)
+
+        """
+
+        self.__update_results(force_update)
+        # {"totalMisses": 6, "totalImpacts": 0, "totalCloseApproaches": 12}
+        misses = self._summary.get('totalMisses')
+        if misses is None:
+            misses = 0
+        close_approaches = self._summary.get('totalCloseApproaches')
+        if close_approaches is None:
+            close_approaches = 0
+        impacts = self._summary.get('totalImpacts')
+        if impacts is None:
+            impacts = 0
+        probability = impacts / (misses + impacts)
```

Comment:
If misses and impacts are both zero, do we want to do anything to handle the divide by zero? This is admittedly a corner case that occurs when there's a problem with the job, but can make for some confusing messaging about divide by zero.